blas/tpls/KokkosBlas1_swap_tpl_spec_decl.hpp

```diff
@@ -0,0 +1,590 @@
+  };
+
+#define KOKKOSBLAS1_CSWAP_TPL_SPEC_DECL_BLAS(LAYOUT, EXECSPACE,                \
+                                             ETI_SPEC_AVAIL)                   \
+  template <>                                                                  \
+  struct Swap<EXECSPACE,                                                       \
+              Kokkos::View<Kokkos::complex<float>*, LAYOUT,                    \
+                           Kokkos::Device<EXECSPACE, Kokkos::HostSpace>,       \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged>>,           \
+              Kokkos::View<Kokkos::complex<float>*, LAYOUT,                    \
+                           Kokkos::Device<EXECSPACE, Kokkos::HostSpace>,       \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged>>,           \
+              true, ETI_SPEC_AVAIL> {                                          \
+    using XVector = Kokkos::View<Kokkos::complex<float>*, LAYOUT,              \
+                                 Kokkos::Device<EXECSPACE, Kokkos::HostSpace>, \
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>;     \
+    using YVector = Kokkos::View<Kokkos::complex<float>*, LAYOUT,              \
+                                 Kokkos::Device<EXECSPACE, Kokkos::HostSpace>, \
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>;     \
+    static void rot(EXECSPACE const& /*space*/, XVector const& X,              \
+                    YVector const& Y) {                                        \
+      Kokkos::Profiling::pushRegion(                                           \
+          "KokkosBlas::swap[TPL_BLAS,complex<float>]");                        \
+      HostBlas<std::complex<float>>::swap(                                     \
+          X.extent_int(0), reinterpret_cast<std::complex<float>*>(X.data()),   \
+          1, reinterpret_cast<std::complex<float>*>(Y.data()), 1);             \
+      Kokkos::Profiling::popRegion();                                          \
+    }                                                                          \
+  };
+
+#ifdef KOKKOS_ENABLE_SERIAL
+KOKKOSBLAS1_DSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, true)
+KOKKOSBLAS1_DSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, false)
+
+KOKKOSBLAS1_SSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, true)
+KOKKOSBLAS1_SSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, false)
+
+KOKKOSBLAS1_ZSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, true)
+KOKKOSBLAS1_ZSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, false)
+
+KOKKOSBLAS1_CSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, true)
+KOKKOSBLAS1_CSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::Serial, false)
+#endif
+
+#ifdef KOKKOS_ENABLE_OPENMP
+KOKKOSBLAS1_DSWAP_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::OpenMP, true)
```

Comment:
Do we always want to replace the native OpenMP impl with BLAS TPL? I worry that the TPL may not always use parallelism, in which case this replaces a parallel impl with a sequential one.